# Gaming Channel Stream Suggestions
Recommendations for streaming channel.

Steam player counts --updated constantly up to May 2022

*https://www.statista.com/statistics/552623/number-games-released-steam/*


In [1]:
import pandas as pd
import numpy as np
from IPython.display import display
pd.set_option("display.max_columns", None)


In [2]:
# Read in the dataframe
# Steam player trends over time
trends = pd.read_csv("steam_charts.csv", na_values = "-") # <-- fills "artificial" nulls with NaN

In [3]:
trends.head()

Month  Avg. Players      Gain  % Gain  Peak Players  App ID  \
0   Last 30 Days     570825.94   5864.20  +1.04%        923996     730   
1       May 2022     564961.70  -4021.46  -0.71%        923996     730   
2     April 2022     568983.16 -12506.53  -2.15%       1013237     730   
3     March 2022     581489.69 -53148.71  -8.37%        987993     730   
4  February 2022     634638.40  32262.13  +5.36%        995163     730   

                               Game  
0  Counter-Strike: Global Offensive  
1  Counter-Strike: Global Offensive  
2  Counter-Strike: Global Offensive  
3  Counter-Strike: Global Offensive  
4  Counter-Strike: Global Offensive

In [4]:
trends.info(memory_usage="deep")

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52284 entries, 0 to 52283
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Month         52284 non-null  object 
 1   Avg. Players  52284 non-null  float64
 2   Gain          51285 non-null  float64
 3   % Gain        51285 non-null  object 
 4   Peak Players  52284 non-null  int64  
 5   App ID        52284 non-null  int64  
 6   Game          52284 non-null  object 
dtypes: float64(2), int64(2), object(3)
memory usage: 12.1 MB


In [5]:
# Percent of missing data
trends.apply(pd.isnull).sum()/trends.shape[0] * 100

Month           0.000000
Avg. Players    0.000000
Gain            1.910718
% Gain          1.910718
Peak Players    0.000000
App ID          0.000000
Game            0.000000
dtype: float64

## Data Cleaning
To do:
1. Convert `Month` to pandas datetime
2. Convert `Gain` to float
3. Convert `% Gain` to float
4. Rename columns
5. Deal with missing data

### Month column

Since this dataset is constantly updated by the author, you can see that the month for June 2022 is called "Last 30 Days". I believe that the player data is updated at the end of the month, since it would be impossible to know the full picture of a monthly trend until the end of the month. Since I am doing this analysis at the end of June 2022, I'm confident that all the data for for at least up to the beginning of June is updated. 

While the pandas `to_datetime` function is quite flexible and can interpret many ways of writing dates as strings, it cannot interpret "Last 30 Days". Therefore I'm going to change that string to "June 2022" so that it will not throw an error. 

This function converts strings to a datetime object with a default of YYYY-MM-DD with the day always being "01". As mentioned above, the data is probably not for the beginning of the month, instead it seems to be updated at the end of the month. I'm going to use `pandas.offsets.MonthEnd()` function to change the day value to the appropriate end of month day (28, 29, 30, or 31).

In [6]:
# String replacement
trends["Month"] = trends["Month"].str.replace("Last 30 Days", "June 2022")

# To datetime function on month column
trends["Month"] = pd.to_datetime(trends["Month"]) + pd.offsets.MonthEnd() # <-- offset to month end instead of beginning of month

In [7]:
# Earliest entry for player data
trends["Month"].min()

Timestamp('2012-07-31 00:00:00')

In [8]:
# Set index to the month column
trends = trends.set_index("Month")

In [9]:
trends

Avg. Players      Gain   % Gain  Peak Players  App ID  \
Month                                                               
2022-06-30     570825.94   5864.20   +1.04%        923996     730   
2022-05-31     564961.70  -4021.46   -0.71%        923996     730   
2022-04-30     568983.16 -12506.53   -2.15%       1013237     730   
2022-03-31     581489.69 -53148.71   -8.37%        987993     730   
2022-02-28     634638.40  32262.13   +5.36%        995163     730   
...                  ...       ...      ...           ...     ...   
2020-06-30        444.60    -37.79   -7.83%          1019  625960   
2020-05-31        482.40   -605.05  -55.64%          1400  625960   
2020-04-30       1087.45    410.17  +60.56%          4339  625960   
2020-03-31        677.28  -2603.64  -79.36%          1992  625960   
2020-02-29       3280.91       NaN      NaN         10706  625960   

                                        Game  
Month                                         
2022-06-30  Counter-Strike: Global Offensive  
2022-05-31  Counter-Strike: Global Offensive  
2022-04-30  Counter-Strike: Global Offensive  
2022-03-31  Counter-Strike: Global Offensive  
2022-02-28  Counter-Strike: Global Offensive  
...                                      ...  
2020-06-30                        Stoneshard  
2020-05-31                        Stoneshard  
2020-04-30                        Stoneshard  
2020-03-31                        Stoneshard  
2020-02-29                        Stoneshard  

[52284 rows x 6 columns]

In [10]:
# Overview of game frequency
trends["Game"].value_counts()

Counter-Strike: Global Offensive      120
Grand Theft Auto: San Andreas         120
SimCity 4 Deluxe                      120
Counter-Strike: Source                120
Mount & Blade: With Fire and Sword    120
                                     ... 
Godsbane Idle                           1
Metal: Hellsinger Demo                  1
Terra Invicta Demo                      1
Builder Simulator                       1
Starship Troopers: Terran Command       1
Name: Game, Length: 997, dtype: int64

In [11]:
# Total number of unique games
trends["Game"].nunique()

997

### Converting columns to float: `Gain` and `% Gain`

Both columns that describe the actual player trends overtime are stored as string objects. It's obvious in `% Gain`, because each value has a percent sign attached at the end. For meaningful analysis these columns are going to be converted to a float.

In [12]:
# Converts gain column to float
trends["Gain"] = trends["Gain"].astype(float)

In [13]:
# Remove "%"
# Cast as float
trends["% Gain"] = trends["% Gain"].str.replace("%","").astype(float)

In [14]:
trends.columns

Index(['Avg. Players', 'Gain', '% Gain', 'Peak Players', 'App ID', 'Game'], dtype='object')

In [15]:
# Column renaming
cols = {
    "Avg. Players":"average_players",
    "Gain":"player_gain",
    "% Gain":"player_gain_percentage",
    "Peak Players":"peak_players",
    "App ID":"app_id",
    "Game":"game"
    }

trends = trends.rename(columns=cols)

In [16]:
# Reset index
trends = trends.reset_index(drop=False)

In [17]:
trends.iloc[1235:].head(30)

Month  average_players  player_gain  player_gain_percentage  \
1235 2013-03-31          1960.13      1960.13             43776179.05   
1236 2013-02-28             0.00         0.00                     inf   
1237 2013-01-31             0.00          NaN                     NaN   
1238 2022-06-30         14668.59      -387.20                   -2.57   
1239 2022-05-31         15055.81      7393.86                   96.50   
1240 2022-04-30          7661.95     -2316.69                  -23.22   
1241 2022-03-31          9978.64    -13536.96                  -57.57   
1242 2022-02-28         23515.60    -51285.27                  -68.56   
1243 2022-01-31         74800.87          NaN                     NaN   
1244 2022-06-30         51382.83     -2149.80                   -4.02   
1245 2022-05-31         53532.61     -3621.31                   -6.34   
1246 2022-04-30         57153.92      1678.76                    3.03   
1247 2022-03-31         55475.16      -366.05                   -0.66   
1248 2022-02-28         55841.21     -3903.22                   -6.53   
1249 2022-01-31         59744.43    -11766.28                  -16.45   
1250 2021-12-31         71510.71     -6621.83                   -8.48   
1251 2021-11-30         78132.54     32073.73                   69.64   
1252 2021-10-31         46058.81     15182.38                   49.17   
1253 2021-09-30         30876.43     19801.91                  178.81   
1254 2021-08-31         11074.52          NaN                     NaN   
1255 2022-06-30         28817.86     -6889.00                  -19.29   
1256 2022-05-31         35706.87      -929.90                   -2.54   
1257 2022-04-30         36636.77    -12494.22                  -25.43   
1258 2022-03-31         49130.98    -37993.22                  -43.61   
1259 2022-02-28         87124.20    -75197.88                  -46.33   
1260 2022-01-31        162322.08          NaN                     NaN   
1261 2022-06-30         35025.43     -2928.00                   -7.71   
1262 2022-05-31         37953.48       603.03                    1.61   
1263 2022-04-30         37350.45      -938.37                   -2.45   
1264 2022-03-31         38288.82     -1055.84                   -2.68   

      peak_players   app_id                         game  
1235         19099   230410                     Warframe  
1236             1   230410                     Warframe  
1237             0   230410                     Warframe  
1238         56219  1446780          MONSTER HUNTER RISE  
1239         31601  1446780          MONSTER HUNTER RISE  
1240         20566  1446780          MONSTER HUNTER RISE  
1241         27679  1446780          MONSTER HUNTER RISE  
1242         61070  1446780          MONSTER HUNTER RISE  
1243        133629  1446780          MONSTER HUNTER RISE  
1244         59697  1623660                         MIR4  
1245         59697  1623660                         MIR4  
1246         62099  1623660                         MIR4  
1247         61468  1623660                         MIR4  
1248         61535  1623660                         MIR4  
1249         71226  1623660                         MIR4  
1250         91672  1623660                         MIR4  
1251         97125  1623660                         MIR4  
1252         66095  1623660                         MIR4  
1253         41962  1623660                         MIR4  
1254         22147  1623660                         MIR4  
1255         61032  1449850       Yu-Gi-Oh!  Master Duel  
1256        102062  1449850       Yu-Gi-Oh!  Master Duel  
1257         86259  1449850       Yu-Gi-Oh!  Master Duel  
1258        119337  1449850       Yu-Gi-Oh!  Master Duel  
1259        190442  1449850       Yu-Gi-Oh!  Master Duel  
1260        262333  1449850       Yu-Gi-Oh!  Master Duel  
1261         55679   289070  Sid Meier's Civilization VI  
1262         61740   289070  Sid Meier's Civilization VI  
1263         66605   289070  Sid Me

### Missing data

Earlier I noticed that the columns `Gain`, `% Gain` (now `player_gain` and `player_gain_percentage`) had slightly less than 2% of the rows missing data.
For whatever reason some of this information is missing, thus we'll use the `.ffill()` method (forward fill) to fill missing data with data from a row that precedes a missing row. This approach is fine for time series data that only has one unique value in a column. However, it would be inaccurate to simply use the forward fill method with this dataset, because it contains nearly 1,000 different games. If I forward fill data from one game to a value that contains a different game in the next row it would produce inaccurate results.

To overcome this flaw pandas has the `groupby()` function. You can first groupby the `game` column, and forward fill missing data in only the `player_gain` and `player_gain_percentage` columns. I will create a new dataframe from this approach and merge it back to the original dataframe on the index of both datasets. One thing I realized when I used the `.ffill()` method was it did not completely handle all missing data. I realized that the remaining rows with incomplete data were all from June 2022. These were all new games that did not have any previous entries in the dataset to be able to forward fill. These rows had to be filled in with 0s, since there could not be a gain or loss to calculate without previous data.

When two dataframes with the same column names are merged, this new dataframe produces column names that are preceded with "_x" or "_y" to distinguish them. I dropped the `player_gain_x` and `player_gain_percetage_x` columns (original, missing data), and kept the new columns. The new columns were renamed to drop the "_x" and "_y" at the end.

In [18]:
# Group by operation by game
# Forward fill on just the player gain/player gain percent column
# Set a new copy
fill_na = trends.groupby("game")[["player_gain","player_gain_percentage"]].ffill().copy()

In [19]:
# Verify if forward fill successful
fill_na[fill_na.isnull().any(axis=1)]

player_gain  player_gain_percentage
1894           NaN                     NaN
5179           NaN                     NaN
6081           NaN                     NaN
6082           NaN                     NaN
6947           NaN                     NaN
7268           NaN                     NaN
7397           NaN                     NaN
7484           NaN                     NaN
8394           NaN                     NaN
9198           NaN                     NaN
11782          NaN                     NaN
14462          NaN                     NaN
15494          NaN                     NaN
18146          NaN                     NaN
18593          NaN                     NaN
18762          NaN                     NaN
19221          NaN                     NaN
19562          NaN                     NaN
22562          NaN                     NaN
26669          NaN                     NaN
27738          NaN                     NaN
28819          NaN                     NaN
34631          NaN                     NaN
37819          NaN                     NaN
38191          NaN                     NaN
38245          NaN                     NaN
40343          NaN                     NaN
44913          NaN                     NaN
47598          NaN                     NaN
50344          NaN                     NaN
51687          NaN                     NaN

In [20]:
# Creates a list of all the indices that have missing data points
index_list = fill_na[fill_na.isnull().any(axis=1)].index.tolist()

In [21]:
# Locates all games from the original trends dataset that have missing data left over from the forward fill method
trends.iloc[index_list]

Month  average_players  player_gain  player_gain_percentage  \
1894  2022-06-30         26371.23          NaN                     NaN   
5179  2022-06-30         10597.80          NaN                     NaN   
6081  2022-06-30          6718.75          NaN                     NaN   
6082  2022-06-30          7996.06          NaN                     NaN   
6947  2022-06-30          6503.37          NaN                     NaN   
7268  2022-06-30          6324.53          NaN                     NaN   
7397  2022-06-30          4627.41          NaN                     NaN   
7484  2022-06-30          2730.76          NaN                     NaN   
8394  2022-06-30          6186.56          NaN                     NaN   
9198  2022-06-30          3443.79          NaN                     NaN   
11782 2022-06-30          3301.96          NaN                     NaN   
14462 2022-06-30          3371.34          NaN                     NaN   
15494 2022-06-30          2124.74          NaN                     NaN   
18146 2022-06-30          2059.90          NaN                     NaN   
18593 2022-06-30          1018.84          NaN                     NaN   
18762 2022-06-30          1632.20          NaN                     NaN   
19221 2022-06-30           601.23          NaN                     NaN   
19562 2022-06-30           766.06          NaN                     NaN   
22562 2022-06-30          1315.18          NaN                     NaN   
26669 2022-06-30           719.75          NaN                     NaN   
27738 2022-06-30           970.33          NaN                     NaN   
28819 2022-06-30           467.89          NaN                     NaN   
34631 2022-06-30           519.58          NaN                     NaN   
37819 2022-06-30           116.90          NaN                     NaN   
38191 2022-06-30           609.30          NaN                     NaN   
38245 2022-06-30           624.20          NaN                     NaN   
40343 2022-06-30           444.86          NaN                     NaN   
44913 2022-06-30           293.37          NaN                     NaN   
47598 2022-06-30           563.03          NaN                     NaN   
50344 2022-06-30           270.37          NaN                     NaN   
51687 2022-06-30           633.58          NaN                     NaN   

       peak_players   app_id                                              game  
1894          38020   868270                               The Cycle: Frontier  
5179          14484  1824220                                        Chivalry 2  
6081          10778  1462040               FINAL FANTASY VII REMAKE INTERGRADE  
6082          12680  1966900                              20 Minutes Till Dawn  
6947           8700  1361510  Teenage Mutant Ninja Turtles: Shredder's Revenge  
7268           8089  1202130                 Starship Troopers: Terran Command  
7397           9667  1836450                Monster Hunter Rise: Sunbreak Demo  
7484           7330  1979310                        VEILED EXPERTS GLOBAL BETA  
8394          14794  1029690                                    Sniper Elite 5  
9198           6125  1594320                               Captain of Industry  
11782         10063  1993130                                          弈仙牌 Demo  
14462          6278  1577120                                        The Quarry  
15494          3175  1488200                Symphony of War: The Nephilim Saga  
18146          2865  1533420                                        Neon White  
18593          2064  1767560                          Pro Cycling Manager 2022  
18762          2086  1565140                                     Godsbane Idle  
19221          1536  2013570                             Cult of the Lamb Demo  
19562          1905  1996930                                          Pacifish  
22562         12666  2023830                                Terra Invicta Demo  
26669          1022  1

In [22]:
fill_na = fill_na.fillna(0)

In [23]:
# Uses indices of missing data 
# All unique game names from this list
# Assigns to new variable
game_list_nulls = trends.iloc[index_list]["game"].unique().tolist()

In [24]:
# Verifies that these games only show up here in June 2022 and no where else
# Games that do not have information for player gain and player gain percentage
# All released in June 2022
trends[trends["game"].isin(game_list_nulls)]

Month  average_players  player_gain  player_gain_percentage  \
1894  2022-06-30         26371.23          NaN                     NaN   
5179  2022-06-30         10597.80          NaN                     NaN   
6081  2022-06-30          6718.75          NaN                     NaN   
6082  2022-06-30          7996.06          NaN                     NaN   
6947  2022-06-30          6503.37          NaN                     NaN   
7268  2022-06-30          6324.53          NaN                     NaN   
7397  2022-06-30          4627.41          NaN                     NaN   
7484  2022-06-30          2730.76          NaN                     NaN   
8394  2022-06-30          6186.56          NaN                     NaN   
9198  2022-06-30          3443.79          NaN                     NaN   
11782 2022-06-30          3301.96          NaN                     NaN   
14462 2022-06-30          3371.34          NaN                     NaN   
15494 2022-06-30          2124.74          NaN                     NaN   
18146 2022-06-30          2059.90          NaN                     NaN   
18593 2022-06-30          1018.84          NaN                     NaN   
18762 2022-06-30          1632.20          NaN                     NaN   
19221 2022-06-30           601.23          NaN                     NaN   
19562 2022-06-30           766.06          NaN                     NaN   
22562 2022-06-30          1315.18          NaN                     NaN   
26669 2022-06-30           719.75          NaN                     NaN   
27738 2022-06-30           970.33          NaN                     NaN   
28819 2022-06-30           467.89          NaN                     NaN   
34631 2022-06-30           519.58          NaN                     NaN   
37819 2022-06-30           116.90          NaN                     NaN   
38191 2022-06-30           609.30          NaN                     NaN   
38245 2022-06-30           624.20          NaN                     NaN   
40343 2022-06-30           444.86          NaN                     NaN   
44913 2022-06-30           293.37          NaN                     NaN   
47598 2022-06-30           563.03          NaN                     NaN   
50344 2022-06-30           270.37          NaN                     NaN   
51687 2022-06-30           633.58          NaN                     NaN   

       peak_players   app_id                                              game  
1894          38020   868270                               The Cycle: Frontier  
5179          14484  1824220                                        Chivalry 2  
6081          10778  1462040               FINAL FANTASY VII REMAKE INTERGRADE  
6082          12680  1966900                              20 Minutes Till Dawn  
6947           8700  1361510  Teenage Mutant Ninja Turtles: Shredder's Revenge  
7268           8089  1202130                 Starship Troopers: Terran Command  
7397           9667  1836450                Monster Hunter Rise: Sunbreak Demo  
7484           7330  1979310                        VEILED EXPERTS GLOBAL BETA  
8394          14794  1029690                                    Sniper Elite 5  
9198           6125  1594320                               Captain of Industry  
11782         10063  1993130                                          弈仙牌 Demo  
14462          6278  1577120                                        The Quarry  
15494          3175  1488200                Symphony of War: The Nephilim Saga  
18146          2865  1533420                                        Neon White  
18593          2064  1767560                          Pro Cycling Manager 2022  
18762          2086  1565140                                     Godsbane Idle  
19221          1536  2013570                             Cult of the Lamb Demo  
19562          1905  1996930                                          Pacifish  
22562         12666  2023830                                Terra Invicta Demo  
26669          1022  1

In [25]:
# Merge dataframes
# Assign as new variable
new_df = pd.merge(trends, fill_na, left_index=True, right_index=True)

In [26]:
new_df

Month  average_players  player_gain_x  player_gain_percentage_x  \
0     2022-06-30        570825.94        5864.20                      1.04   
1     2022-05-31        564961.70       -4021.46                     -0.71   
2     2022-04-30        568983.16      -12506.53                     -2.15   
3     2022-03-31        581489.69      -53148.71                     -8.37   
4     2022-02-28        634638.40       32262.13                      5.36   
...          ...              ...            ...                       ...   
52279 2020-06-30           444.60         -37.79                     -7.83   
52280 2020-05-31           482.40        -605.05                    -55.64   
52281 2020-04-30          1087.45         410.17                     60.56   
52282 2020-03-31           677.28       -2603.64                    -79.36   
52283 2020-02-29          3280.91            NaN                       NaN   

       peak_players  app_id                              game  player_gain_y  \
0            923996     730  Counter-Strike: Global Offensive        5864.20   
1            923996     730  Counter-Strike: Global Offensive       -4021.46   
2           1013237     730  Counter-Strike: Global Offensive      -12506.53   
3            987993     730  Counter-Strike: Global Offensive      -53148.71   
4            995163     730  Counter-Strike: Global Offensive       32262.13   
...             ...     ...                               ...            ...   
52279          1019  625960                        Stoneshard         -37.79   
52280          1400  625960                        Stoneshard        -605.05   
52281          4339  625960                        Stoneshard         410.17   
52282          1992  625960                        Stoneshard       -2603.64   
52283         10706  625960                        Stoneshard       -2603.64   

       player_gain_percentage_y  
0                          1.04  
1                         -0.71  
2                         -2.15  
3                         -8.37  
4                          5.36  
...                         ...  
52279                     -7.83  
52280                    -55.64  
52281                     60.56  
52282                    -79.36  
52283                    -79.36  

[52284 rows x 9 columns]

In [27]:
new_df.columns

Index(['Month', 'average_players', 'player_gain_x', 'player_gain_percentage_x',
       'peak_players', 'app_id', 'game', 'player_gain_y',
       'player_gain_percentage_y'],
      dtype='object')

In [28]:
# Drop original columns

# List of columns
cols_to_drop = ['player_gain_x', 'player_gain_percentage_x']

# Drop columns
new_df = new_df.drop(cols_to_drop, axis=1)

In [29]:
# Column renaming

# Dictionary of columns to rename
cols_to_rename = {
    "player_gain_y":"player_gain",
    "player_gain_percentage_y":"player_gain_percentage"
    }

# Rename, set index to month, and set copy
# New dataset is "trends_clean"
trends_clean = new_df.rename(columns=cols_to_rename).copy()

In [30]:
# Verify cleaning process
trends_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52284 entries, 0 to 52283
Data columns (total 7 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Month                   52284 non-null  datetime64[ns]
 1   average_players         52284 non-null  float64       
 2   peak_players            52284 non-null  int64         
 3   app_id                  52284 non-null  int64         
 4   game                    52284 non-null  object        
 5   player_gain             52284 non-null  float64       
 6   player_gain_percentage  52284 non-null  float64       
dtypes: datetime64[ns](1), float64(3), int64(2), object(1)
memory usage: 2.8+ MB


## Changelog for `steam_charts.csv` data

July 2, 2022

### Changes
    - Converted "Month" to datetime YYYY-MM-DD
    - Converted "Gain" to float
    - Converted "% Gain" to float
    - Renamed columns (including "Gain" and "% Gain")

### Fixes
    - Used forward fill method to impute missing data points in "player_gain" and "player_gain_percentage" columns
    - Any remaining missing data replaced with 0s

In [31]:
# Create new csv from cleaned dataframe
# Run only ONCE
#trends_clean.to_csv("steam_charts_clean.csv", index=False)

## Additional data
Data that explains player trends over a period of time is useful, but it can be supplemented with extra data. There is extra data for steam games with user reviews, user playtime, descriptions, and other variables (columns) that may prove useful for analysis.

First thing to do is read in the csv files and preview each dataframe with the `.head()` method. I'll use a left merge to keep the original dataset length (52,284 rows). Valve, the owner and developer of the Steam client provides a unique ID for each game, and it is consistent throughout all the datasets provided. This will be the column to perform the merging on to preserve the integrity of the data.

I'm going to be performing the merging one at a time. Following any merge I'll be using `.head()`,`.info()`, or `.sample()` methods to verify there are not any errors or inaccuracies. 

In [117]:
# Additional dataframes 
games = pd.read_csv("steam.csv")
description = pd.read_csv("steam_description_data.csv")
media_data = pd.read_csv("steam_media_data.csv")
optional = pd.read_csv("steam_optional.csv")
packages = pd.read_csv("steam_packages_info.csv")
requirements = pd.read_csv("steam_requirements_data.csv")
support = pd.read_csv("steam_support_info.csv")
tags = pd.read_csv("steamspy_tag_data.csv")


In [132]:
# List of csv files assigned as variables
df_list = [games, description, media_data, optional, packages, requirements, support, tags]

# List of the variable names as strings
df_name_list = ["games:", "description:", "media_data:", "optional:", "packages:", "requirements:", "support:", "tags:"]

# Iterate through both lists using zip to display first 5 rows of each dataframe
# Each item from the "df_name_list" will printed above each dataframe

# Iterate through lists, zip
for item, name in zip(df_list, df_name_list):
    print("\n") # spaces out dataframes
    print(name) # print the name of each dataframe
    display(item.head()) # utilize ipython import to display data



games:


appid  type                       name  required_age  dlc fullgame  \
0     10  game             Counter-Strike             0  NaN      NaN   
1     20  game      Team Fortress Classic             0  NaN      NaN   
2     30  game              Day of Defeat             0  NaN      NaN   
3     40  game         Deathmatch Classic             0  NaN      NaN   
4     50  game  Half-Life: Opposing Force             0  NaN      NaN   

                                 supported_languages            developers  \
0  ['English', 'French', 'German', 'Italian', 'Ko...             ['Valve']   
1  ['English', 'French', 'German', 'Italian', 'Ko...             ['Valve']   
2  ['English', 'French', 'German', 'Italian', 'Sp...             ['Valve']   
3  ['English', 'French', 'German', 'Italian', 'Ko...             ['Valve']   
4          ['English', 'French', 'German', 'Korean']  ['Gearbox Software']   

  publishers     packages                    platforms  \
0  ['Valve']  [574941, 7]  ['windows', 'mac', 'linux']   
1  ['Valve']         [29]  ['windows', 'mac', 'linux']   
2  ['Valve']         [30]  ['windows', 'mac', 'linux']   
3  ['Valve']         [31]  ['windows', 'mac', 'linux']   
4  ['Valve']         [32]  ['windows', 'mac', 'linux']   

                                          categories      genres  \
0  ['Multi-player', 'PvP', 'Online PvP', 'Shared/...  ['Action']   
1  ['Multi-player', 'PvP', 'Online PvP', 'Shared/...  ['Action']   
2       ['Multi-player', 'Valve Anti-Cheat enabled']  ['Action']   
3  ['Multi-player', 'PvP', 'Online PvP', 'Shared/...  ['Action']   
4  ['Single-player', 'Multi-player', 'Valve Anti-...  ['Action']   

   achievements release_date  \
0           0.0   2000-11-01   
1           0.0   1999-04-01   
2           0.0   2003-05-01   
3           0.0   2001-06-01   
4           0.0   1999-11-01   

                                     supported_audio  coming_soon  price  \
0  ['English', 'French', 'German', 'Italian', 'Ko...        False   8.19   
1                                                NaN        False   3.99   
2                                                NaN        False   3.99   
3                                                NaN        False   3.99   
4                                                NaN        False   3.99   

   review_score  total_positive  total_negative     rating             owners  \
0           9.0        117261.0          3686.0  95.566768  10000000-20000000   
1           8.0          3896.0           705.0  81.939532    2000000-5000000   
2           8.0          2794.0           398.0  84.223637   5000000-10000000   
3           6.0          1214.0           308.0  76.485571   5000000-10000000   
4           9.0         11343.0           519.0  92.916082   5000000-10000000   

   average_forever  median_forever  \
0            10499             202   
1             1637              23   
2              169              11   
3             2632               6   
4              442             250   

                                                tags  
0  ['Action', 'FPS', 'Multiplayer', 'Shooter', 'C...  
1  ['Action', 'FPS', 'Multiplayer', 'Classic', 'H...  
2  ['FPS', 'World War II', 'Multiplayer', 'Shoote...  
3  ['Action', 'FPS', 'Classic', 'Multiplayer', 'S...  
4  ['FPS', 'Action', 'Classic', 'Sci-fi', 'Single...



description:


appid                               detailed_description  \
0     10  Play the world's number 1 online action game. ...   
1     20  One of the most popular online action games of...   
2     30  Enlist in an intense brand of Axis vs. Allied ...   
3     40  Enjoy fast-paced multiplayer gaming with Death...   
4     50  Return to the Black Mesa Research Facility as ...   

                                      about_the_game  \
0  Play the world's number 1 online action game. ...   
1  One of the most popular online action games of...   
2  Enlist in an intense brand of Axis vs. Allied ...   
3  Enjoy fast-paced multiplayer gaming with Death...   
4  Return to the Black Mesa Research Facility as ...   

                                   short_description  
0  Play the world's number 1 online action game. ...  
1  One of the most popular online action games of...  
2  Enlist in an intense brand of Axis vs. Allied ...  
3  Enjoy fast-paced multiplayer gaming with Death...  
4  Return to the Black Mesa Research Facility as ...



media_data:


appid                                       header_image  \
0     10  https://cdn.akamai.steamstatic.com/steam/apps/...   
1     20  https://cdn.akamai.steamstatic.com/steam/apps/...   
2     30  https://cdn.akamai.steamstatic.com/steam/apps/...   
3     40  https://cdn.akamai.steamstatic.com/steam/apps/...   
4     50  https://cdn.akamai.steamstatic.com/steam/apps/...   

                                         screenshots  \
0  [{'id': 0, 'path_thumbnail': 'https://cdn.akam...   
1  [{'id': 0, 'path_thumbnail': 'https://cdn.akam...   
2  [{'id': 0, 'path_thumbnail': 'https://cdn.akam...   
3  [{'id': 0, 'path_thumbnail': 'https://cdn.akam...   
4  [{'id': 0, 'path_thumbnail': 'https://cdn.akam...   

                                          background movies  
0  https://cdn.akamai.steamstatic.com/steam/apps/...    NaN  
1  https://cdn.akamai.steamstatic.com/steam/apps/...    NaN  
2  https://cdn.akamai.steamstatic.com/steam/apps/...    NaN  
3  https://cdn.akamai.steamstatic.com/steam/apps/...    NaN  
4  https://cdn.akamai.steamstatic.com/steam/apps/...    NaN



optional:


appid drm_notice ext_user_account_notice demos  \
0     10        NaN                     NaN   NaN   
1     20        NaN                     NaN   NaN   
2     30        NaN                     NaN   NaN   
3     70        NaN                     NaN   NaN   
4     80        NaN                     NaN   NaN   

                    content_descriptors  metacritic_score  \
0  Includes intense violence and blood.              88.0   
1  Includes intense violence and blood.               NaN   
2                                   NaN              79.0   
3                                   NaN              96.0   
4                                   NaN              65.0   

                                      metacritic_url  
0  https://www.metacritic.com/game/pc/counter-str...  
1                                                NaN  
2  https://www.metacritic.com/game/pc/day-of-defe...  
3  https://www.metacritic.com/game/pc/half-life?f...  
4  https://www.metacritic.com/game/pc/counter-str...



packages:


appid     type                          title  is_recurring_subscription  \
0     10  default             Buy Counter-Strike                      False   
1     20  default      Buy Team Fortress Classic                      False   
2     30  default              Buy Day of Defeat                      False   
3     40  default         Buy Deathmatch Classic                      False   
4     50  default  Buy Half-Life: Opposing Force                      False   

                                                subs  
0  [{'packageid': 7, 'percent_savings_text': ' ',...  
1  [{'packageid': 29, 'percent_savings_text': ' '...  
2  [{'packageid': 30, 'percent_savings_text': ' '...  
3  [{'packageid': 31, 'percent_savings_text': ' '...  
4  [{'packageid': 32, 'percent_savings_text': ' '...



requirements:


appid                                    pc_requirements  \
0     10  {'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...   
1     20  {'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...   
2     30  {'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...   
3     40  {'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...   
4     50  {'minimum': '\r\n\t\t\t<p><strong>Minimum:</st...   

                                    mac_requirements  \
0  {'minimum': 'Minimum: OS X  Snow Leopard 10.6....   
1  {'minimum': 'Minimum: OS X  Snow Leopard 10.6....   
2  {'minimum': 'Minimum: OS X  Snow Leopard 10.6....   
3  {'minimum': 'Minimum: OS X  Snow Leopard 10.6....   
4  {'minimum': 'Minimum: OS X  Snow Leopard 10.6....   

                                  linux_requirements  \
0  {'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...   
1  {'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...   
2  {'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...   
3  {'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...   
4  {'minimum': 'Minimum: Linux Ubuntu 12.04, Dual...   

                                          pc_minimum pc_recommended  \
0  500 mhz processor, 96mb ram, 16mb video card, ...            NaN   
1  500 mhz processor, 96mb ram, 16mb video card, ...            NaN   
2  500 mhz processor, 96mb ram, 16mb video card, ...            NaN   
3  500 mhz processor, 96mb ram, 16mb video card, ...            NaN   
4  500 mhz processor, 96mb ram, 16mb video card, ...            NaN   

                                         mac_minimum mac_recommended  
0  OS X  Snow Leopard 10.6.3, 1GB RAM, 4GB Hard D...             NaN  
1  OS X  Snow Leopard 10.6.3, 1GB RAM, 4GB Hard D...             NaN  
2  OS X  Snow Leopard 10.6.3, 1GB RAM, 4GB Hard D...             NaN  
3  OS X  Snow Leopard 10.6.3, 1GB RAM, 4GB Hard D...             NaN  
4  OS X  Snow Leopard 10.6.3, 1GB RAM, 4GB Hard D...             NaN



support:


appid                      website                       support_url  \
0     10                          NaN  http://steamcommunity.com/app/10   
1     30  http://www.dayofdefeat.com/                               NaN   
2     50                          NaN     https://help.steampowered.com   
3     70    http://www.half-life.com/  http://steamcommunity.com/app/70   
4     80                          NaN  http://steamcommunity.com/app/80   

  support_email  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN



tags:


appid  1980s  1990s  2.5d  2d  2d_fighter  2d_platformer  360_video  3d  \
0     10    262   1185     0   0           0              0          0   0   
1     20      0    138     0   0           0              0          0   0   
2     30      0      0     0   0           0              0          0   0   
3     40      0      9     0   0           0              0          0   0   
4     50      0    137     0   0           0              0          0   0   

   3d_fighter  3d_platformer  3d_vision  4_player_local  4x  6dof  \
0           0              0          0               0   0     0   
1           0              0          0               0   0     0   
2           0              0          0               0   0     0   
3           0              0          0               0   0     0   
4           0              0          0               0   0     0   

   8_bit_music  atv  abstract  action  action_rpg  action_rts  \
0            0    0         0    5391           0           0   
1            0    0         0     749           0           0   
2            0    0         0     160           0           0   
3            0    0         0     631           0           0   
4            0    0         0     328           0           0   

   action_roguelike  action_adventure  addictive  adventure  agriculture  \
0                 0                 0          0          0            0   
1                 0                 0          0          0            0   
2                 0                 0          0          0            0   
3                 0                 0          0          0            0   
4                 0                 0          0        119            0   

   aliens  alternate_history  ambient  america  animation_&_modeling  anime  \
0       0                  0        0        0                     0      0   
1       0                  0        0        0                     0      0   
2       0                  0        0        0                     0      0   
3       0                  0        0        0                     0      0   
4     178                  0        0        0                     0      0   

   arcade  archery  arena_shooter  artificial_intelligence  assassin  \
0       0        0              0                        0       225   
1       0        0              0                        0         0   
2       0        0              0                        0         0   
3       0        0             46                        0         0   
4       0        0              0                        0         0   

   asymmetric_vr  asynchronous_multiplayer  atmospheric  audio_production  \
0              0                         0            0                 0   
1              0                         0            0                 0   
2              0                         0            0                 0   
3              0                         0            0                 0   
4              0                         0          107                 0   

   auto_battler  automation  automobile_sim  bmx  base_building  baseball  \
0             0           0               0    0              0         0   
1             0           0               0    0              0         0   
2             0           0               0    0              0         0   
3             0           0               0    0              0         0   
4             0           0               0    0              0         0   

   based_on_a_novel  basketball  batman  battle_royale  beat_em_up  beautiful  \
0                 0           0       0              0           0          0   
1                 0           0       0              0           0          0   
2                 0           0       0              0           0          0   
3                 0           0       0              0           0          0   
4                 0           0 

### Dataframes to use

1. description: description of each game
2. games: majority of numeric data to include price, release date, rating, etc..
3. optional: some extra descriptions, and includes metacritic score data for each app
4. packages: most useful column is "is_recurring_subscription"

The Kaggle page for these datasets were contained in one zip folder when I downloaded the datasets. As seen above, the author supplied more datasets, but I believe these four datasets will suffice for analysis.

During the process of merging dataframes I noticed that I kept ending up with extra rows. The maximum dataframe length should not have exceeded 52,284 rows. This happened after merging the `packages` dataframe, because this set of data had duplicated game IDs, thus it would create extra rows. To fix this problem I used the pandas method `.drop_duplicates()` with the subset set to `appid` on the `packages` dataframe before merging to the main dataframe `trends_clean`.

In [35]:
games.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 102504 entries, 0 to 102503
Data columns (total 26 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   appid                102504 non-null  int64  
 1   type                 102504 non-null  object 
 2   name                 102504 non-null  object 
 3   required_age         102504 non-null  int64  
 4   dlc                  9696 non-null    object 
 5   fullgame             34607 non-null   object 
 6   supported_languages  102352 non-null  object 
 7   developers           102463 non-null  object 
 8   publishers           102464 non-null  object 
 9   packages             81153 non-null   object 
 10  platforms            102504 non-null  object 
 11  categories           102398 non-null  object 
 12  genres               102311 non-null  object 
 13  achievements         102504 non-null  float64
 14  release_date         95676 non-null   object 
 15  supported_audio  

In [86]:
# Frequency of type (game, dlc, ads, music) in games dataframe
games["type"].value_counts()

game           67870
dlc            34632
advertising        1
music              1
Name: type, dtype: int64

In [38]:
#trends = pd.read_csv("steam_charts_clean.csv")

In [39]:
# Merging with description dataframe
trends_clean = trends_clean.merge(description, how="left",left_on="app_id", right_on="appid")

In [40]:
trends_clean.head()

Month  average_players  peak_players  app_id  \
0 2022-06-30        570825.94        923996     730   
1 2022-05-31        564961.70        923996     730   
2 2022-04-30        568983.16       1013237     730   
3 2022-03-31        581489.69        987993     730   
4 2022-02-28        634638.40        995163     730   

                               game  player_gain  player_gain_percentage  \
0  Counter-Strike: Global Offensive      5864.20                    1.04   
1  Counter-Strike: Global Offensive     -4021.46                   -0.71   
2  Counter-Strike: Global Offensive    -12506.53                   -2.15   
3  Counter-Strike: Global Offensive    -53148.71                   -8.37   
4  Counter-Strike: Global Offensive     32262.13                    5.36   

   appid                               detailed_description  \
0  730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
1  730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
2  730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
3  730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
4  730.0  Counter-Strike: Global Offensive (CS: GO) expa...   

                                      about_the_game  \
0  Counter-Strike: Global Offensive (CS: GO) expa...   
1  Counter-Strike: Global Offensive (CS: GO) expa...   
2  Counter-Strike: Global Offensive (CS: GO) expa...   
3  Counter-Strike: Global Offensive (CS: GO) expa...   
4  Counter-Strike: Global Offensive (CS: GO) expa...   

                                   short_description  
0  Counter-Strike: Global Offensive (CS: GO) expa...  
1  Counter-Strike: Global Offensive (CS: GO) expa...  
2  Counter-Strike: Global Offensive (CS: GO) expa...  
3  Counter-Strike: Global Offensive (CS: GO) expa...  
4  Counter-Strike: Global Offensive (CS: GO) expa...

In [41]:
trends_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52284 entries, 0 to 52283
Data columns (total 11 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Month                   52284 non-null  datetime64[ns]
 1   average_players         52284 non-null  float64       
 2   peak_players            52284 non-null  int64         
 3   app_id                  52284 non-null  int64         
 4   game                    52284 non-null  object        
 5   player_gain             52284 non-null  float64       
 6   player_gain_percentage  52284 non-null  float64       
 7   appid                   47466 non-null  float64       
 8   detailed_description    47466 non-null  object        
 9   about_the_game          47466 non-null  object        
 10  short_description       47466 non-null  object        
dtypes: datetime64[ns](1), float64(4), int64(2), object(4)
memory usage: 4.8+ MB


In [42]:
# Merging with games dataframe
trends_clean = trends_clean.merge(games, how="left",left_on="app_id", right_on="appid")

In [43]:
trends_clean.head()

Month  average_players  peak_players  app_id  \
0 2022-06-30        570825.94        923996     730   
1 2022-05-31        564961.70        923996     730   
2 2022-04-30        568983.16       1013237     730   
3 2022-03-31        581489.69        987993     730   
4 2022-02-28        634638.40        995163     730   

                               game  player_gain  player_gain_percentage  \
0  Counter-Strike: Global Offensive      5864.20                    1.04   
1  Counter-Strike: Global Offensive     -4021.46                   -0.71   
2  Counter-Strike: Global Offensive    -12506.53                   -2.15   
3  Counter-Strike: Global Offensive    -53148.71                   -8.37   
4  Counter-Strike: Global Offensive     32262.13                    5.36   

   appid_x                               detailed_description  \
0    730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
1    730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
2    730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
3    730.0  Counter-Strike: Global Offensive (CS: GO) expa...   
4    730.0  Counter-Strike: Global Offensive (CS: GO) expa...   

                                      about_the_game  \
0  Counter-Strike: Global Offensive (CS: GO) expa...   
1  Counter-Strike: Global Offensive (CS: GO) expa...   
2  Counter-Strike: Global Offensive (CS: GO) expa...   
3  Counter-Strike: Global Offensive (CS: GO) expa...   
4  Counter-Strike: Global Offensive (CS: GO) expa...   

                                   short_description  appid_y  type  \
0  Counter-Strike: Global Offensive (CS: GO) expa...    730.0  game   
1  Counter-Strike: Global Offensive (CS: GO) expa...    730.0  game   
2  Counter-Strike: Global Offensive (CS: GO) expa...    730.0  game   
3  Counter-Strike: Global Offensive (CS: GO) expa...    730.0  game   
4  Counter-Strike: Global Offensive (CS: GO) expa...    730.0  game   

                               name  required_age        dlc fullgame  \
0  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
1  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
2  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
3  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
4  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   

                                 supported_languages  \
0  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
1  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
2  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
3  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
4  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   

                               developers publishers                 packages  \
0  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
1  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
2  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
3  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
4  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   

                     platforms  \
0  ['windows', 'mac', 'linux']   
1  ['windows', 'mac', 'linux']   
2  ['windows', 'mac', 'linux']   
3  ['windows', 'mac', 'linux']   
4  ['windows', 'mac', 'linux']   

                                          categories  \
0  ['Multi-player', 'Steam Achievements', 'Full c...   
1  ['Multi-player', 'Steam Achievements', 'Full c...   
2  ['Multi-player', 'Steam Achievements', 'Full c...   
3  ['Multi-player', 'Steam Achievements', 'Full c...   
4  ['Multi-player', 'Steam Achievements', 'Full c...   

                       genres  achievements release_date supported_audio  \
0  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
1  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
2  ['Action', 'Free to Play']  

In [44]:
trends_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52284 entries, 0 to 52283
Data columns (total 37 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Month                   52284 non-null  datetime64[ns]
 1   average_players         52284 non-null  float64       
 2   peak_players            52284 non-null  int64         
 3   app_id                  52284 non-null  int64         
 4   game                    52284 non-null  object        
 5   player_gain             52284 non-null  float64       
 6   player_gain_percentage  52284 non-null  float64       
 7   appid_x                 47466 non-null  float64       
 8   detailed_description    47466 non-null  object        
 9   about_the_game          47466 non-null  object        
 10  short_description       47466 non-null  object        
 11  appid_y                 47466 non-null  float64       
 12  type                    47466 non-null  object

The `.sample()` method returns random rows. You can see that games in original dataset (`trends_clean`) are matching up correctly with the additional datasets that were merged to it. If you search the game on the Steam store it will display the categories, developers and publishers which correctly match here in this new dataframe.

In [45]:
trends_clean.sample(10)

Month  average_players  peak_players   app_id  \
25881 2022-01-31          1592.97          2826   571740   
44991 2016-01-31           404.86          1325   232890   
50658 2014-11-30           615.36          1145   234330   
7492  2021-11-30          7109.91         12818   444090   
18312 2020-07-31          3772.18          6436   629760   
35368 2020-08-31           403.97           767   753640   
10454 2018-02-28           614.45          1033   233860   
42289 2022-02-28           777.95          1783  1235140   
45285 2017-12-31           832.26          2222      400   
15303 2019-04-30          2072.56          3775   211820   

                        game  player_gain  player_gain_percentage    appid_x  \
25881               Golf It!       297.56                   22.97   571740.0   
44991  Stronghold Crusader 2        36.57                    9.93   232890.0   
50658    Marvel Puzzle Quest       -42.23                   -6.42   234330.0   
7492                Paladins      -466.20                   -6.15   444090.0   
18312                MORDHAU        21.62                    0.58   629760.0   
35368            Outer Wilds      -605.20                  -59.97   753640.0   
10454                 Kenshi        37.88                    6.57   233860.0   
42289  Yakuza: Like a Dragon      -231.77                  -22.95  1235140.0   
45285                 Portal       312.09                   60.00      400.0   
15303              Starbound      -154.27                   -6.93   211820.0   

                                    detailed_description  \
25881  <strong>Golf It!</strong> is a multiplayer Min...   
44991  <h1>Digital Deluxe Edition</h1><p><img src="ht...   
50658  Marvel Comics Super Heroes come to life in thi...   
7492   <h1>FREE Team Fortress 2 Barik Skin</h1><p><st...   
18312  <strong>MORDHAU</strong> is a medieval first &...   
35368  Winner of Best Game at the 2020 BAFTA Games Aw...   
10454  <a href="https://steamcommunity.com/linkfilter...   
42289  <img src="https://cdn.akamai.steamstatic.com/s...   
45285  <p>Portal&trade; is a new single player game f...   
15303  <h1>Bounty Hunter Update</h1><p>In our latest ...   

                                          about_the_game  \
25881  <strong>Golf It!</strong> is a multiplayer Min...   
44991  <img src="https://cdn.akamai.steamstatic.com/s...   
50658  Marvel Comics Super Heroes come to life in thi...   
7492   <img src="https://cdn.akamai.steamstatic.com/s...   
18312  <strong>MORDHAU</strong> is a medieval first &...   
35368  Winner of Best Game at the 2020 BAFTA Games Aw...   
10454  <a href="https://steamcommunity.com/linkfilter...   
42289  <img src="https://cdn.akamai.steamstatic.com/s...   
45285  <p>Portal&trade; is a new single player game f...   
15303  <img src="https://cdn.akamai.steamstatic.com/s...   

                                       short_description    appid_y  type  \
25881  Golf It! is a multiplayer Minigolf game with f...   571740.0  game   
44991  Stronghold Crusader 2 is the long awaited sequ...   232890.0  game   
50658  The best in Match 3 puzzle gameplay meets Marv...   234330.0  game   
7492   Join 50+ million players in Paladins, the free...   444090.0  game   
18312  MORDHAU is a multiplayer medieval slasher. Cre...   629760.0  game   
35368  Named Game of the Year 2019 by Giant Bomb, Pol...   753640.0  game   
10454  A free-roaming squad based RPG. Focusing on op...   233860.0  game   
42289  Become Ichiban Kasuga, a low-ranking yakuza gr...  1235140.0  game   
45285  Portal&trade; is a new single player game from...      400.0  game   
15303  You’ve fled your home, only to find yourself l...   211820.0  game   

                        name  required_age  \
25881               Golf It!           0.0   
44991  Stronghold Crusader 2           0.0   
50658    MARVEL Puzzle Quest           0.0   
7492               Paladins®           0.0   
18312                MORDHAU           0.0   
35368            Outer 

In [48]:
# Merging with "optional" dataframe
trends_clean = trends_clean.merge(optional, how="left",left_on="app_id", right_on="appid")

In [49]:
# Displays dataframe columns only
trends_clean.columns

Index(['Month', 'average_players', 'peak_players', 'app_id', 'game',
       'player_gain', 'player_gain_percentage', 'appid_x',
       'detailed_description', 'about_the_game', 'short_description',
       'appid_y', 'type', 'name', 'required_age', 'dlc', 'fullgame',
       'supported_languages', 'developers', 'publishers', 'packages',
       'platforms', 'categories', 'genres', 'achievements', 'release_date',
       'supported_audio', 'coming_soon', 'price', 'review_score',
       'total_positive', 'total_negative', 'rating', 'owners',
       'average_forever', 'median_forever', 'tags', 'appid', 'drm_notice',
       'ext_user_account_notice', 'demos', 'content_descriptors',
       'metacritic_score', 'metacritic_url'],
      dtype='object')

I'll drop some columns that I know we won't need. Some of them are duplicated "appid" columns that are produced after merging data. Further below there will be more data cleaning.

In [50]:
# List of some columns to remove
columns_to_drop = ['appid_x','appid_y','appid','drm_notice','ext_user_account_notice']

# Drop columns
trends_clean = trends_clean.drop(columns=columns_to_drop)

In [51]:
trends_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52284 entries, 0 to 52283
Data columns (total 39 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   Month                   52284 non-null  datetime64[ns]
 1   average_players         52284 non-null  float64       
 2   peak_players            52284 non-null  int64         
 3   app_id                  52284 non-null  int64         
 4   game                    52284 non-null  object        
 5   player_gain             52284 non-null  float64       
 6   player_gain_percentage  52284 non-null  float64       
 7   detailed_description    47466 non-null  object        
 8   about_the_game          47466 non-null  object        
 9   short_description       47466 non-null  object        
 10  type                    47466 non-null  object        
 11  name                    47466 non-null  object        
 12  required_age            47466 non-null  float6

In [61]:
# This is a preview of how the "packages" data is adding extra rows
trends_clean.merge(packages, how="left",left_on="app_id", right_on="appid").tail()

Month  average_players  peak_players  app_id  \
0     2022-06-30        570825.94        923996     730   
1     2022-05-31        564961.70        923996     730   
2     2022-04-30        568983.16       1013237     730   
3     2022-03-31        581489.69        987993     730   
4     2022-02-28        634638.40        995163     730   
...          ...              ...           ...     ...   
53367 2020-06-30           444.60          1019  625960   
53368 2020-05-31           482.40          1400  625960   
53369 2020-04-30          1087.45          4339  625960   
53370 2020-03-31           677.28          1992  625960   
53371 2020-02-29          3280.91         10706  625960   

                                   game  player_gain  player_gain_percentage  \
0      Counter-Strike: Global Offensive      5864.20                    1.04   
1      Counter-Strike: Global Offensive     -4021.46                   -0.71   
2      Counter-Strike: Global Offensive    -12506.53                   -2.15   
3      Counter-Strike: Global Offensive    -53148.71                   -8.37   
4      Counter-Strike: Global Offensive     32262.13                    5.36   
...                                 ...          ...                     ...   
53367                        Stoneshard       -37.79                   -7.83   
53368                        Stoneshard      -605.05                  -55.64   
53369                        Stoneshard       410.17                   60.56   
53370                        Stoneshard     -2603.64                  -79.36   
53371                        Stoneshard     -2603.64                  -79.36   

                                    detailed_description  \
0      Counter-Strike: Global Offensive (CS: GO) expa...   
1      Counter-Strike: Global Offensive (CS: GO) expa...   
2      Counter-Strike: Global Offensive (CS: GO) expa...   
3      Counter-Strike: Global Offensive (CS: GO) expa...   
4      Counter-Strike: Global Offensive (CS: GO) expa...   
...                                                  ...   
53367  <h1>Development Roadmap</h1><p><img src="https...   
53368  <h1>Development Roadmap</h1><p><img src="https...   
53369  <h1>Development Roadmap</h1><p><img src="https...   
53370  <h1>Development Roadmap</h1><p><img src="https...   
53371  <h1>Development Roadmap</h1><p><img src="https...   

                                          about_the_game  \
0      Counter-Strike: Global Offensive (CS: GO) expa...   
1      Counter-Strike: Global Offensive (CS: GO) expa...   
2      Counter-Strike: Global Offensive (CS: GO) expa...   
3      Counter-Strike: Global Offensive (CS: GO) expa...   
4      Counter-Strike: Global Offensive (CS: GO) expa...   
...                                                  ...   
53367  <img src="https://cdn.akamai.steamstatic.com/s...   
53368  <img src="https://cdn.akamai.steamstatic.com/s...   
53369  <img src="https://cdn.akamai.steamstatic.com/s...   
53370  <img src="https://cdn.akamai.steamstatic.com/s...   
53371  <img src="https://cdn.akamai.steamstatic.com/s...   

                                       short_description type_x  \
0      Counter-Strike: Global Offensive (CS: GO) expa...   game   
1      Counter-Strike: Global Offensive (CS: GO) expa...   game   
2      Counter-Strike: Global Offensive (CS: GO) expa...   game   
3      Counter-Strike: Global Offensive (CS: GO) expa...   game   
4      Counter-Strike: Global Offensive (CS: GO) expa...   game   
...                                                  ...    ...   
53367  Stoneshard is a challenging turn-based RPG set...   game   
53368  Stoneshard is a challenging turn-based RPG set...   game   
53369  Stoneshard is a challenging turn-based RPG set...   game   
53370  Stoneshard is a challenging turn-based RPG set...   game   
53371  Stoneshard is a challenging turn-based RPG set...   game   

                                   name  required_age        dlc fullgame  \
0      Counter-St

In [63]:
# Drop duplicates from the "appid" column in the packages dataframe
packages = packages.drop_duplicates(subset=["appid"])

In [65]:
# Merging with games dataframe
trends_clean = trends_clean.merge(packages, how="left",left_on="app_id", right_on="appid")

In [67]:
# Final dataframe info
# Memory usage over 500mb
trends_clean.info(memory_usage="deep")

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52284 entries, 0 to 52283
Data columns (total 44 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   Month                      52284 non-null  datetime64[ns]
 1   average_players            52284 non-null  float64       
 2   peak_players               52284 non-null  int64         
 3   app_id                     52284 non-null  int64         
 4   game                       52284 non-null  object        
 5   player_gain                52284 non-null  float64       
 6   player_gain_percentage     52284 non-null  float64       
 7   detailed_description       47466 non-null  object        
 8   about_the_game             47466 non-null  object        
 9   short_description          47466 non-null  object        
 10  type_x                     47466 non-null  object        
 11  name                       47466 non-null  object        
 12  requ

## Further data cleaning to be performed

The player trends (`steam_charts.csv`) data was cleaned earlier. I wanted to get that accomplished before merging more data, because it's easier to work with only a few columns in one dataset rather than trying to wait till merging multiple datasets with dozens of columns.

Sometimes when merging data you end up with missing data or unneccesary columns. Before creating a new csv file from this new data some of these nulls, and extra columns need to be dealt with.

Below you can use the following code to show all missing data for each column in descending order. Typically, I scrutinize columns with over 50% missing data. The first three highest columns will be dropped because they contain over 90% missing values.



In [70]:
# Percent of missing data, sorted descending order
(trends_clean.apply(pd.isnull).sum()/trends_clean.shape[0] * 100).sort_values(ascending=False)

fullgame                     100.000000
content_descriptors           92.976819
demos                         91.500268
metacritic_score              50.418866
metacritic_url                50.418866
dlc                           42.869712
supported_audio               41.840716
subs                          23.521536
appid                         23.521536
title                         23.521536
is_recurring_subscription     23.521536
type_y                        23.521536
packages                      22.850203
price                         10.205799
tags                           9.968633
release_date                   9.874914
genres                         9.656874
total_positive                 9.215056
review_score                   9.215056
rating                         9.215056
total_negative                 9.215056
owners                         9.215056
coming_soon                    9.215056
median_forever                 9.215056
average_forever                9.215056


In [72]:
trends_clean.columns

Index(['Month', 'average_players', 'peak_players', 'app_id', 'game',
       'player_gain', 'player_gain_percentage', 'detailed_description',
       'about_the_game', 'short_description', 'type_x', 'name', 'required_age',
       'dlc', 'fullgame', 'supported_languages', 'developers', 'publishers',
       'packages', 'platforms', 'categories', 'genres', 'achievements',
       'release_date', 'supported_audio', 'coming_soon', 'price',
       'review_score', 'total_positive', 'total_negative', 'rating', 'owners',
       'average_forever', 'median_forever', 'tags', 'demos',
       'content_descriptors', 'metacritic_score', 'metacritic_url', 'appid',
       'type_y', 'title', 'is_recurring_subscription', 'subs'],
      dtype='object')

In [81]:
# Show the difference between the different descriptions
trends_clean[['detailed_description','about_the_game', 'short_description']].sample(10)

detailed_description  \
43003  <img src="https://cdn.akamai.steamstatic.com/s...   
19425  Sid Meier's Civilization IV®: Beyond the Sword...   
37208  <h2 class="bb_tag">The Classic Adventure Conti...   
45214                                                NaN   
3911   The Flagship Turn-Based Strategy Game Returns<...   
50653  Marvel Comics Super Heroes come to life in thi...   
39934  Play the top-rated tower defense franchise in ...   
27211  <h1>GOLD EDITION</h1><p>The Gold Edition inclu...   
45101  Gang Beasts is a silly multiplayer party game ...   
41542  <img src="https://cdn.akamai.steamstatic.com/s...   

                                          about_the_game  \
43003  <img src="https://cdn.akamai.steamstatic.com/s...   
19425  Sid Meier's Civilization IV®: Beyond the Sword...   
37208  <h2 class="bb_tag">The Classic Adventure Conti...   
45214                                                NaN   
3911   The Flagship Turn-Based Strategy Game Returns<...   
50653  Marvel Comics Super Heroes come to life in thi...   
39934  Play the top-rated tower defense franchise in ...   
27211  London, 1868. In the heart of the Industrial R...   
45101  Gang Beasts is a silly multiplayer party game ...   
41542  <img src="https://cdn.akamai.steamstatic.com/s...   

                                       short_description  
43003  100% Orange Juice is a digital multiplayer boa...  
19425  Sid Meier's Civilization IV®: Beyond the Sword...  
37208  Rediscover the beloved RPG classic— now enhanc...  
45214                                                NaN  
3911   Create, discover, and download new player-crea...  
50653  The best in Match 3 puzzle gameplay meets Marv...  
39934  Go head to head with other players in a Bloon-...  
27211  London, 1868. In the heart of the Industrial R...  
45101  Gang Beasts is a silly multiplayer party game ...  
41542  A massively multiplayer grand strategy game, S...

Setting the random state to "12" for `.sample()` shows how merging on the `app_id` was an appropriate choice. The game names are the same, however you can see how they may be spelled slightly differently.

For example, "Sid Meier's Civilization IV: Beyond the Sword" is the same game as "Civilization IV: Beyond the Sword". They have the same app ID but different datasets add or remove words or use different punctuation. Not merging on the ID would have produced too many inconsistencies.

I spent several minutes reviewing the columns to determine if any more needed to be removed. While having extra data is great, more data does not automatically mean it's useful. Without going into too much detail for each variable or column, the ones that will be dropped below are not going to be relevant to the analysis. They're mostly redundent columns or may contain a single value that does not add anything new.

In [105]:
# Game names match but there are differences
trends_clean[["game","name"]].sample(10,random_state=12)

game  \
5390                                        Soundpad   
19346  Sid Meier's Civilization IV: Beyond the Sword   
13068                        A Dance of Fire and Ice   
38022               The Elder Scrolls III: Morrowind   
21967                         FTL: Faster Than Light   
13158                                       Crossout   
14856      Grand Theft Auto IV: The Complete Edition   
35426                                          UBOAT   
35388                           She Will Punish Them   
50215                                    Bejeweled 3   

                                                    name  
5390                                                 NaN  
19346                  Civilization IV: Beyond the Sword  
13068                            A Dance of Fire and Ice  
38022  The Elder Scrolls III: Morrowind® Game of the ...  
21967                             FTL: Faster Than Light  
13158                                           Crossout  
14856              Grand Theft Auto IV: Complete Edition  
35426                                              UBOAT  
35388                               She Will Punish Them  
50215                                       Bejeweled® 3

In [82]:
trends_clean.head()

Month  average_players  peak_players  app_id  \
0 2022-06-30        570825.94        923996     730   
1 2022-05-31        564961.70        923996     730   
2 2022-04-30        568983.16       1013237     730   
3 2022-03-31        581489.69        987993     730   
4 2022-02-28        634638.40        995163     730   

                               game  player_gain  player_gain_percentage  \
0  Counter-Strike: Global Offensive      5864.20                    1.04   
1  Counter-Strike: Global Offensive     -4021.46                   -0.71   
2  Counter-Strike: Global Offensive    -12506.53                   -2.15   
3  Counter-Strike: Global Offensive    -53148.71                   -8.37   
4  Counter-Strike: Global Offensive     32262.13                    5.36   

                                detailed_description  \
0  Counter-Strike: Global Offensive (CS: GO) expa...   
1  Counter-Strike: Global Offensive (CS: GO) expa...   
2  Counter-Strike: Global Offensive (CS: GO) expa...   
3  Counter-Strike: Global Offensive (CS: GO) expa...   
4  Counter-Strike: Global Offensive (CS: GO) expa...   

                                      about_the_game  \
0  Counter-Strike: Global Offensive (CS: GO) expa...   
1  Counter-Strike: Global Offensive (CS: GO) expa...   
2  Counter-Strike: Global Offensive (CS: GO) expa...   
3  Counter-Strike: Global Offensive (CS: GO) expa...   
4  Counter-Strike: Global Offensive (CS: GO) expa...   

                                   short_description type_x  \
0  Counter-Strike: Global Offensive (CS: GO) expa...   game   
1  Counter-Strike: Global Offensive (CS: GO) expa...   game   
2  Counter-Strike: Global Offensive (CS: GO) expa...   game   
3  Counter-Strike: Global Offensive (CS: GO) expa...   game   
4  Counter-Strike: Global Offensive (CS: GO) expa...   game   

                               name  required_age        dlc fullgame  \
0  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
1  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
2  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
3  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   
4  Counter-Strike: Global Offensive           0.0  [1766730]      NaN   

                                 supported_languages  \
0  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
1  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
2  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
3  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
4  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   

                               developers publishers                 packages  \
0  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
1  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
2  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
3  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   
4  ['Valve', 'Hidden Path Entertainment']  ['Valve']  [329385, 298963, 54029]   

                     platforms  \
0  ['windows', 'mac', 'linux']   
1  ['windows', 'mac', 'linux']   
2  ['windows', 'mac', 'linux']   
3  ['windows', 'mac', 'linux']   
4  ['windows', 'mac', 'linux']   

                                          categories  \
0  ['Multi-player', 'Steam Achievements', 'Full c...   
1  ['Multi-player', 'Steam Achievements', 'Full c...   
2  ['Multi-player', 'Steam Achievements', 'Full c...   
3  ['Multi-player', 'Steam Achievements', 'Full c...   
4  ['Multi-player', 'Steam Achievements', 'Full c...   

                       genres  achievements release_date supported_audio  \
0  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
1  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
2  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
3  ['Action', 'Free to Play']         167.0   2012-08-21 

In [107]:
# More columns to drop
list_of_cols = [
    "fullgame",
    "content_descriptors",
    "demos",
    "appid",
    "type_y",
    "type_x",
    "detailed_description",
    "about_the_game",
    "packages",
    "fullgame",
    "subs",
    "title",
    "name"
    ]

trends_clean = trends_clean.drop(list_of_cols, axis = 1)

In [108]:
trends_clean.head()

Month  average_players  peak_players  app_id  \
0 2022-06-30        570825.94        923996     730   
1 2022-05-31        564961.70        923996     730   
2 2022-04-30        568983.16       1013237     730   
3 2022-03-31        581489.69        987993     730   
4 2022-02-28        634638.40        995163     730   

                               game  player_gain  player_gain_percentage  \
0  Counter-Strike: Global Offensive      5864.20                    1.04   
1  Counter-Strike: Global Offensive     -4021.46                   -0.71   
2  Counter-Strike: Global Offensive    -12506.53                   -2.15   
3  Counter-Strike: Global Offensive    -53148.71                   -8.37   
4  Counter-Strike: Global Offensive     32262.13                    5.36   

                                   short_description  required_age        dlc  \
0  Counter-Strike: Global Offensive (CS: GO) expa...           0.0  [1766730]   
1  Counter-Strike: Global Offensive (CS: GO) expa...           0.0  [1766730]   
2  Counter-Strike: Global Offensive (CS: GO) expa...           0.0  [1766730]   
3  Counter-Strike: Global Offensive (CS: GO) expa...           0.0  [1766730]   
4  Counter-Strike: Global Offensive (CS: GO) expa...           0.0  [1766730]   

                                 supported_languages  \
0  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
1  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
2  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
3  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
4  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   

                               developers publishers  \
0  ['Valve', 'Hidden Path Entertainment']  ['Valve']   
1  ['Valve', 'Hidden Path Entertainment']  ['Valve']   
2  ['Valve', 'Hidden Path Entertainment']  ['Valve']   
3  ['Valve', 'Hidden Path Entertainment']  ['Valve']   
4  ['Valve', 'Hidden Path Entertainment']  ['Valve']   

                     platforms  \
0  ['windows', 'mac', 'linux']   
1  ['windows', 'mac', 'linux']   
2  ['windows', 'mac', 'linux']   
3  ['windows', 'mac', 'linux']   
4  ['windows', 'mac', 'linux']   

                                          categories  \
0  ['Multi-player', 'Steam Achievements', 'Full c...   
1  ['Multi-player', 'Steam Achievements', 'Full c...   
2  ['Multi-player', 'Steam Achievements', 'Full c...   
3  ['Multi-player', 'Steam Achievements', 'Full c...   
4  ['Multi-player', 'Steam Achievements', 'Full c...   

                       genres  achievements release_date supported_audio  \
0  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
1  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
2  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
3  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   
4  ['Action', 'Free to Play']         167.0   2012-08-21    [' English']   

  coming_soon  price  review_score  total_positive  total_negative     rating  \
0       False    0.0           8.0       2949363.0        439344.0  86.634302   
1       False    0.0           8.0       2949363.0        439344.0  86.634302   
2       False    0.0           8.0       2949363.0        439344.0  86.634302   
3       False    0.0           8.0       2949363.0        439344.0  86.634302   
4       False    0.0           8.0       2949363.0        439344.0  86.634302   

               owners  average_forever  median_forever  \
0  50000000-100000000          30053.0          6104.0   
1  50000000-100000000          30053.0          6104.0   
2  50000000-100000000          30053.0          6104.0   
3  50000000-100000000          30053.0          6104.0   
4  50000000-100000000          30053.0          6104.0   

                                                tags  metacritic_score  \
0  ['FPS', 'Shooter', 'Multiplayer', 'Competitive...              83.0   
1  ['FPS', 'Shooter', 'Multiplayer', 'Competitive...              83.0   
2

The author of this data states in their description on Kaggle that the price of the games are in Euros. Valve is an American company, but the author is probably from Europe, thus when they collected the data it was in their local currency.

The author last updated the collection of datasets on May 25, 2022, therefore I will be using the conversion rate of that day.

**Wednesday 25, May 2022 conversion:**

**€1 EUR = $1.0689 US dollar**

I'll finalize the *majority* of the data cleaning by converting the price to US dollars, and renaming a few columns.

In [110]:
#Wednesday 25 May 2022	€1 EUR = $1.0689
trends_clean["price"] = trends_clean["price"] * 1.0689

In [113]:
# Dictionary of columns to rename
col_rename = {
    "price":"price_usd",
    "Month":"month",
    }

# Rename columns
combined_df_trends = trends_clean.rename(columns=col_rename).copy() # <-- Set copy

In [114]:
# Combined dataframes to a single csv file
#combined_df_trends.to_csv("steam_charts_combined.csv",index=False)

In [115]:
pd.read_csv("steam_charts_combined.csv")

month  average_players  peak_players  app_id  \
0      2022-06-30        570825.94        923996     730   
1      2022-05-31        564961.70        923996     730   
2      2022-04-30        568983.16       1013237     730   
3      2022-03-31        581489.69        987993     730   
4      2022-02-28        634638.40        995163     730   
...           ...              ...           ...     ...   
52279  2020-06-30           444.60          1019  625960   
52280  2020-05-31           482.40          1400  625960   
52281  2020-04-30          1087.45          4339  625960   
52282  2020-03-31           677.28          1992  625960   
52283  2020-02-29          3280.91         10706  625960   

                                   game  player_gain  player_gain_percentage  \
0      Counter-Strike: Global Offensive      5864.20                    1.04   
1      Counter-Strike: Global Offensive     -4021.46                   -0.71   
2      Counter-Strike: Global Offensive    -12506.53                   -2.15   
3      Counter-Strike: Global Offensive    -53148.71                   -8.37   
4      Counter-Strike: Global Offensive     32262.13                    5.36   
...                                 ...          ...                     ...   
52279                        Stoneshard       -37.79                   -7.83   
52280                        Stoneshard      -605.05                  -55.64   
52281                        Stoneshard       410.17                   60.56   
52282                        Stoneshard     -2603.64                  -79.36   
52283                        Stoneshard     -2603.64                  -79.36   

                                       short_description  required_age  \
0      Counter-Strike: Global Offensive (CS: GO) expa...           0.0   
1      Counter-Strike: Global Offensive (CS: GO) expa...           0.0   
2      Counter-Strike: Global Offensive (CS: GO) expa...           0.0   
3      Counter-Strike: Global Offensive (CS: GO) expa...           0.0   
4      Counter-Strike: Global Offensive (CS: GO) expa...           0.0   
...                                                  ...           ...   
52279  Stoneshard is a challenging turn-based RPG set...           0.0   
52280  Stoneshard is a challenging turn-based RPG set...           0.0   
52281  Stoneshard is a challenging turn-based RPG set...           0.0   
52282  Stoneshard is a challenging turn-based RPG set...           0.0   
52283  Stoneshard is a challenging turn-based RPG set...           0.0   

             dlc                                supported_languages  \
0      [1766730]  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
1      [1766730]  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
2      [1766730]  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
3      [1766730]  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
4      [1766730]  ['Bulgarian', 'Czech', 'Danish', 'Dutch', 'Eng...   
...          ...                                                ...   
52279  [1230230]  ['English', 'French', 'German', 'Italian', 'Ja...   
52280  [1230230]  ['English', 'French', 'German', 'Italian', 'Ja...   
52281  [1230230]  ['English', 'French', 'German', 'Italian', 'Ja...   
52282  [1230230]  ['English', 'French', 'German', 'Italian', 'Ja...   
52283  [1230230]  ['English', 'French', 'German', 'Italian', 'Ja...   

                                   developers             publishers  \
0      ['Valve', 'Hidden Path Entertainment']              ['Valve']   
1      ['Valve', 'Hidden Path Entertainment']              ['Valve']   
2      ['Valve', 'Hidden Path Entertainment']              ['Valve']   
3      ['Valve', 'Hidden Path Entertainment']              ['Valve']   
4      ['Valve', 'Hidden Path Entertainment']              ['Valve']   
...                                       ...                    ...   
52279                    ['Ink Stains Games']  ['HypeTrain Digital']   
52280       